iohub/ngff/models.py

```diff
@@ -156,22 +183,25 @@ class TransformationMeta(MetaBase):
     # MUST
     type: Literal["identity", "translation", "scale"]
     # MUST? (keyword not found in spec for the fields below)
-    translation: Optional[list[float]] = None
-    scale: Optional[list[float]] = None
-    path: Optional[str] = None
-
-    @root_validator
-    def no_extra_method(cls, values: dict):
-        count = sum([bool(v) for _, v in values.items()])
-        if values["type"] == "identity" and count > 1:
+    translation: list[float] | None = None
+    scale: list[float] | None = None
+    path: Annotated[str, Field(min_length=1)] | None = None
+
+    @model_validator(mode="after")
+    def no_extra_method(self) -> Self:
+        methods = sum(
+            bool(m is not None)
+            for m in [self.translation, self.scale, self.path]
+        )
+        if self.type == "identity" and methods > 0:
             raise ValueError(
                 "Method should not be specified for identity transformation!"
             )
-        elif count > 2:
+        elif self.translation and self.scale:
             raise ValueError(
-                "Only one type of transformation method is allowed."
+                "Translation cannot be set to be both translation and scale."
```

Comment:
```suggestion
                "'translation' and 'scale' cannot be provided in the same 'TransformationMeta'"
```
I think this might be more clear.